### PR TITLE
ticdc: add old value feature and column bit flag feature

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

+ Add the old value feature and bit flags of columns https://github.com/pingcap/docs-cn/pull/4209
+ Update open protocol tables https://github.com/pingcap/docs-cn/pull/4355
+ Update the note about a bug https://github.com/pingcap/docs-cn/pull/4234

The TiCDC architecture image is updated in https://github.com/pingcap/docs/pull/3779.

The reviewer of this PR can directly refer to the latest upstream version.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4209 https://github.com/pingcap/docs-cn/pull/4355 https://github.com/pingcap/docs-cn/pull/4234 https://github.com/pingcap/docs-cn/pull/4413
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: ticdc/ticdc-open-protocol.md [hunk 1]
```diff
@@ -66,17 +66,77 @@ This section introduces the formats of Row Changed Event, DDL Event, and Resolve
...
+                "t":<Column Type>,
+                "h":<Where Handle>,
+                "f":<Flag>,
+                "v":<Column Value>
+            }
+        }
+    }
+    ```
+
+    `Delete` event. The deleted row data is output. When the old value feature is enabled, the `Delete` event includes all the rows of the deleted row data; when this feature is disabled, the `Delete` event only includes the [HandleKey](#bit-flags-of-columns) row.
```
Comment (ran-huang) [inline ticdc/ticdc-open-protocol.md:125]:
```suggestion
    `Delete` event. The deleted row data is output. When the old value feature is enabled, the `Delete` event includes all the columns of the deleted row data; when this feature is disabled, the `Delete` event only includes the [HandleKey](#bit-flags-of-columns) column.
```

File: ticdc/ticdc-open-protocol.md [hunk 2]
```diff
@@ -210,44 +270,41 @@ COMMIT;
...
+| BIT                   | 16   | {"t":16,"v":81} | |
+| JSON                  | 245  | {"t":245,"v":"{\\"key1\\": \\"value1\\"}"} | |
+| DECIMAL               | 246  | {"t":246,"v":"129012.1230000"} | |
+| ENUM                  | 247  | {"t":247,"v":1} | |
+| SET                   | 248  | {"t":248,"v":3} | |
+| TINTTEXT/TINTBLOB     | 249  | {"t":249,"v":"5rWL6K+VdGV4dA=="} | The value is encoded in Base64. |
+| MEDIUMTEXT/MEDIUMBLOB | 250  | {"t":250,"v":"5rWL6K+VdGV4dA=="} | The value is encoded in Base64. |
+| LONGTEXT/LONGBLOB     | 251  | {"t":251,"v":"5rWL6K+VdGV4dA=="} | The value is encoded in Base64. |
+| TEXT/BLOB             | 252  | {"t":252,"v":"5rWL6K+VdGV4dA=="} | The value is encoded in Base64. |
+| CHAR/BINARY           | 254  | {"t":254,"v":"test"} / {"t":254,"v":"\\x89PNG\\r\\n\\x1a\\n"} | The value is encoded in UTF-8. When the upstream type is VARBINARY, invisible ASCII characters are escaped. |
```
Comment (ran-huang) [inline ticdc/ticdc-open-protocol.md:302]:
```suggestion
| CHAR/BINARY           | 254  | {"t":254,"v":"test"} / {"t":254,"v":"\\x89PNG\\r\\n\\x1a\\n"} | The value is encoded in UTF-8. When the upstream type is BINARY, invisible ASCII characters are escaped. |
```

File: ticdc/ticdc-open-protocol.md [hunk 3]
```diff
@@ -287,3 +344,41 @@ The column and DDL type codes are encodings defined by TiCDC Open Protocol. `Col
 | Create Sequence                   | 34 |
 | Alter Sequence                    | 35 |
 | Drop Sequence                     | 36 |
+
+## Bit flags of columns
+
+The bit flags represents specific attributes of columns.
```
Comment (ran-huang) [inline ticdc/ticdc-open-protocol.md:350]:
```suggestion
The bit flags represent specific attributes of columns.
```

File: ticdc/ticdc-open-protocol.md [hunk 4]
```diff
@@ -287,3 +344,41 @@ The column and DDL type codes are encodings defined by TiCDC Open Protocol. `Col
...
+46 == 0b_010_1110
+   == MultipleKeyFlag | PrimaryKeyFlag | GeneratedColumnFlag | HandleKeyFlag
+```
+
+> **Note:**
+>
+> + This feature is still experimental. Do **NOT** use it in the production environment.
+> + Currently, for a known composite index column, the `UniqueKeyFlag` value of this column is incorrectly output.
+> + `BinaryFlag` is meaningful only when the column type is Blob/Text (including Tiny Blob/Tiny Text and Long Blob/Long Text). When the upstream column is the Blob type, the `BinaryFlag` value is set to `1`. When the upstream column is the Text type, the `BinaryFlag` value is set to `0`.
+> + To replicate a table from the upstream, TiCDC selects a [valid index](/ticdc/ticdc-overview.md#restrictions) as the Handle index. The `HandleKeyFlag` value of the Handle index column is set `1`.
```
Comment (ran-huang) [inline ticdc/ticdc-open-protocol.md:384]:
```suggestion
> + To replicate a table from the upstream, TiCDC selects a [valid index](/ticdc/ticdc-overview.md#restrictions) as the Handle index. The `HandleKeyFlag` value of the Handle index column is set to `1`.
```

File: ticdc/ticdc-open-protocol.md [hunk 5]
```diff
@@ -66,17 +66,77 @@ This section introduces the formats of Row Changed Event, DDL Event, and Resolve
 
 + **Value:**
 
+    `Insert` event. Newly added data is output.
```
Comment (amyangfei) [inline ticdc/ticdc-open-protocol.md:69]:
```suggestion
    `Insert` event. The newly added row data is output.
```

File: ticdc/ticdc-open-protocol.md [hunk 6]
```diff
@@ -287,3 +344,41 @@ The column and DDL type codes are encodings defined by TiCDC Open Protocol. `Col
...
+
+```
+46 == 0b_010_1110
+   == MultipleKeyFlag | PrimaryKeyFlag | GeneratedColumnFlag | HandleKeyFlag
+```
+
+> **Note:**
+>
+> + This feature is still experimental. Do **NOT** use it in the production environment.
+> + Currently, for a known composite index column, the `UniqueKeyFlag` value of this column is incorrectly output.
```
Comment (amyangfei) [inline ticdc/ticdc-open-protocol.md:382]: Seems this bug has been fixed, I will confirm it later.

Comment (amyangfei) [reply ticdc/ticdc-open-protocol.md:382]: I have confirmed this bug has been fixed, we can remove L382. Besides the Chinese document also needs to update.

Comment (amyangfei) [reply ticdc/ticdc-open-protocol.md:382]:
```suggestion
```

**Review by amyangfei — COMMENTED** [review]
LGTM

File: ticdc/ticdc-overview.md [hunk 7]
```diff
@@ -70,10 +70,12 @@ Currently, the TiCDC sink component supports replicating data to the following d
 
 ## Restrictions
 
-To replicate data to TiDB or MySQL, you must ensure that the following requirements are satisfied to guarantee data correctness:
+TiCDC only replicates the table that has at least one **valid index**. **valid index** is defined as follows
```
Comment (ran-huang) [inline ticdc/ticdc-overview.md:73]:
```suggestion
TiCDC only replicates the table that has at least one **valid index**. A **valid index** is defined as follows:
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

**Pull request merged**